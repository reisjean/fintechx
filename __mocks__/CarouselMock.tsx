import { useEffect } from 'react'
import { FaArrowLeftLong, FaArrowRightLong } from 'react-icons/fa6'

export const CarouselMock = jest.fn(({ children, setApi }) => {
  useEffect(() => {
    const api = {
      scrollSnapList: jest.fn(() => [0, 1]),
      selectedScrollSnap: jest.fn(() => 0),
      on: jest.fn((event, cb) => {
        if (event === 'select') {
          cb()
        }
      }),
    }

    setApi(api)
  }, [setApi])
  return <div>{children}</div>
})

export const CarouselContentMock = jest.fn(({ children }) => <div>{children}</div>)

export const CarouselItemMock = jest.fn(({ children }) => <div>{children}</div>)

export const CarouselNextMock = jest.fn(() =>
  <button>
    <FaArrowRightLong />
    <span className="sr-only">Next slide</span>
  </button>
)

export const CarouselPreviousMock = jest.fn(() =>
  <button>
    <FaArrowLeftLong />
    <span className="sr-only">Previous slide</span>
  </button>
)
