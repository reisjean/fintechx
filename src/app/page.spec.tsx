import { render, screen } from '@testing-library/react'
import { CarouselContentMock, CarouselItemMock, CarouselMock, CarouselNextMock, CarouselPreviousMock } from '__mocks__/CarouselMock'
import Home from './page'

jest.mock("../components/ui/carousel", () => ({
  Carousel: CarouselMock,
  CarouselContent: CarouselContentMock,
  CarouselItem: CarouselItemMock,
  CarouselNext: CarouselNextMock,
  CarouselPrevious: CarouselPreviousMock,
}))
describe('Page', () => {
  it('renders a heading', () => {
    const homePage = render(<Home />)

    expect(screen.getByRole('button', { name: 'Skip' })).toBeInTheDocument()
  })
})