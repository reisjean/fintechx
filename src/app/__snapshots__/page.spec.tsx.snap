// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders a heading 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="text-center  flex flex-col items-center"
      >
        <div
          class="text-right w-full px-6 pt-2 mb-[-1rem]"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 underline-offset-4 hover:underline h-9 px-4 py-2 text-grey-2 text-lg className font-semibold"
          >
            Skip
          </button>
        </div>
        <div
          class="relative w-full"
        >
          <div>
            <div>
              <div>
                <div
                  class="mb-[-1rem] w-full"
                >
                  <img
                    alt="Robot"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="554"
                    sizes="100vw"
                    src="/_next/image?url=%2Frobot.png&w=3840&q=75"
                    srcset="/_next/image?url=%2Frobot.png&w=640&q=75 640w, /_next/image?url=%2Frobot.png&w=750&q=75 750w, /_next/image?url=%2Frobot.png&w=828&q=75 828w, /_next/image?url=%2Frobot.png&w=1080&q=75 1080w, /_next/image?url=%2Frobot.png&w=1200&q=75 1200w, /_next/image?url=%2Frobot.png&w=1920&q=75 1920w, /_next/image?url=%2Frobot.png&w=2048&q=75 2048w, /_next/image?url=%2Frobot.png&w=3840&q=75 3840w"
                    style="color: transparent; width: 100%; height: auto;"
                    width="414"
                  />
                </div>
                <div
                  class="space-y-2"
                >
                  <h1
                    class="className text-neutrals-2 dark:text-white font-bold text-3xl leading-relaxed"
                  >
                    Unlock the Power
                    <br />
                    Of Future AI
                  </h1>
                  <p
                    class="text-paragraph leading-relaxed"
                  >
                    Chat with the smartest AI Future
                    <br />
                    Experience power of AI with us
                  </p>
                </div>
              </div>
              <div>
                <div
                  class="mt-[1.5rem] w-full absolute -z-10"
                >
                  <img
                    alt="Space"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="554"
                    sizes="100vw"
                    src="/_next/image?url=%2Fslide-2.webp&w=3840&q=75"
                    srcset="/_next/image?url=%2Fslide-2.webp&w=640&q=75 640w, /_next/image?url=%2Fslide-2.webp&w=750&q=75 750w, /_next/image?url=%2Fslide-2.webp&w=828&q=75 828w, /_next/image?url=%2Fslide-2.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide-2.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide-2.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide-2.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide-2.webp&w=3840&q=75 3840w"
                    style="color: transparent; width: 100%; height: auto;"
                    width="414"
                  />
                </div>
                <div
                  class="space-y-2 absolute top-3/4 left-1/2 -translate-x-1/2 w-8/12"
                >
                  <h1
                    class="className text-neutrals-2 dark:text-white font-bold text-3xl leading-relaxed"
                  >
                    Harness the Innovation of Next-Gen AI
                  </h1>
                </div>
              </div>
            </div>
            <div
              class="w-full flex items-center justify-center my-6"
            >
              <div
                class="w-[130px] bg-neutrals-8 dark:bg-boxBgDark rounded-elg flex justify-between py-2 px-2 text-neutrals-2 dark:text-white items-center"
              >
                <button>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 288 480 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-370.7 0 73.4-73.4c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-128 128z"
                    />
                  </svg>
                  <span
                    class="sr-only"
                  >
                    Previous slide
                  </span>
                </button>
                <span
                  class="scale-y-150 origin-center"
                >
                  |
                </span>
                <button>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128z"
                    />
                  </svg>
                  <span
                    class="sr-only"
                  >
                    Next slide
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class="flex gap-2 items-center justify-between absolute top-0 left-1/2 -translate-x-1/2 mt-[470px]"
          >
            <div
              class=""
              style="opacity: 0;"
            >
              <svg
                class="text-white"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm0,192a88,88,0,1,1,88-88A88.1,88.1,0,0,1,128,216Zm56-88a56,56,0,1,1-56-56A56.06,56.06,0,0,1,184,128Z"
                />
              </svg>
            </div>
            <div
              class="w-[9px] h-[9px] bg-white rounded-full"
              style="opacity: 0;"
            />
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="text-center  flex flex-col items-center"
    >
      <div
        class="text-right w-full px-6 pt-2 mb-[-1rem]"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 underline-offset-4 hover:underline h-9 px-4 py-2 text-grey-2 text-lg className font-semibold"
        >
          Skip
        </button>
      </div>
      <div
        class="relative w-full"
      >
        <div>
          <div>
            <div>
              <div
                class="mb-[-1rem] w-full"
              >
                <img
                  alt="Robot"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="554"
                  sizes="100vw"
                  src="/_next/image?url=%2Frobot.png&w=3840&q=75"
                  srcset="/_next/image?url=%2Frobot.png&w=640&q=75 640w, /_next/image?url=%2Frobot.png&w=750&q=75 750w, /_next/image?url=%2Frobot.png&w=828&q=75 828w, /_next/image?url=%2Frobot.png&w=1080&q=75 1080w, /_next/image?url=%2Frobot.png&w=1200&q=75 1200w, /_next/image?url=%2Frobot.png&w=1920&q=75 1920w, /_next/image?url=%2Frobot.png&w=2048&q=75 2048w, /_next/image?url=%2Frobot.png&w=3840&q=75 3840w"
                  style="color: transparent; width: 100%; height: auto;"
                  width="414"
                />
              </div>
              <div
                class="space-y-2"
              >
                <h1
                  class="className text-neutrals-2 dark:text-white font-bold text-3xl leading-relaxed"
                >
                  Unlock the Power
                  <br />
                  Of Future AI
                </h1>
                <p
                  class="text-paragraph leading-relaxed"
                >
                  Chat with the smartest AI Future
                  <br />
                  Experience power of AI with us
                </p>
              </div>
            </div>
            <div>
              <div
                class="mt-[1.5rem] w-full absolute -z-10"
              >
                <img
                  alt="Space"
                  data-nimg="1"
                  decoding="async"
                  fetchpriority="high"
                  height="554"
                  sizes="100vw"
                  src="/_next/image?url=%2Fslide-2.webp&w=3840&q=75"
                  srcset="/_next/image?url=%2Fslide-2.webp&w=640&q=75 640w, /_next/image?url=%2Fslide-2.webp&w=750&q=75 750w, /_next/image?url=%2Fslide-2.webp&w=828&q=75 828w, /_next/image?url=%2Fslide-2.webp&w=1080&q=75 1080w, /_next/image?url=%2Fslide-2.webp&w=1200&q=75 1200w, /_next/image?url=%2Fslide-2.webp&w=1920&q=75 1920w, /_next/image?url=%2Fslide-2.webp&w=2048&q=75 2048w, /_next/image?url=%2Fslide-2.webp&w=3840&q=75 3840w"
                  style="color: transparent; width: 100%; height: auto;"
                  width="414"
                />
              </div>
              <div
                class="space-y-2 absolute top-3/4 left-1/2 -translate-x-1/2 w-8/12"
              >
                <h1
                  class="className text-neutrals-2 dark:text-white font-bold text-3xl leading-relaxed"
                >
                  Harness the Innovation of Next-Gen AI
                </h1>
              </div>
            </div>
          </div>
          <div
            class="w-full flex items-center justify-center my-6"
          >
            <div
              class="w-[130px] bg-neutrals-8 dark:bg-boxBgDark rounded-elg flex justify-between py-2 px-2 text-neutrals-2 dark:text-white items-center"
            >
              <button>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 288 480 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-370.7 0 73.4-73.4c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-128 128z"
                  />
                </svg>
                <span
                  class="sr-only"
                >
                  Previous slide
                </span>
              </button>
              <span
                class="scale-y-150 origin-center"
              >
                |
              </span>
              <button>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128z"
                  />
                </svg>
                <span
                  class="sr-only"
                >
                  Next slide
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="flex gap-2 items-center justify-between absolute top-0 left-1/2 -translate-x-1/2 mt-[470px]"
        >
          <div
            class=""
            style="opacity: 0;"
          >
            <svg
              class="text-white"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm0,192a88,88,0,1,1,88-88A88.1,88.1,0,0,1,128,216Zm56-88a56,56,0,1,1-56-56A56.06,56.06,0,0,1,184,128Z"
              />
            </svg>
          </div>
          <div
            class="w-[9px] h-[9px] bg-white rounded-full"
            style="opacity: 0;"
          />
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
